src/resolver.ts

```diff
@@ -32,12 +48,97 @@ export class Resolver {
     this.parseResolveResult = opts.parseResolveResult;
     this.transformDereferenceResult = opts.transformDereferenceResult;
     this.ctx = opts.ctx;
+    this.graph = new DepGraph<string>({ circular: true });
+  }
+
+  public printRefTree(node: string) {
```

Comment:
Suggest pulling this function out into a different file (`tree.ts`?) and breaking it up into smaller parts that we can create tests for.

Reply:
Maybe, I just don’t know if this is something the resolver should do.

Reply:
Meaning have it in this repo in general or wether it should be on that class?